src/tests/ftest/launch.py

```diff
@@ -614,13 +622,27 @@ def get_nvme_replacement(args):
         print("ERROR: Missing a test_servers list to auto-detect NVMe devices")
         sys.exit(1)
 
-    # Get a list of NVMe devices from each specified server host
+    # Get a list of NVMe, VMD devices from each specified server host
+    # if nvme:auto is set, all PCI Non-volatile memory controller
+    # devices are considered NVME (some NVMEs might be behind the VMD
+    # controller). if nvme:vmd, then ignore the NVME
+    # PCI addressed devices behind the VMD controller.
     host_list = list(args.test_servers)
+    include_vmd_flag = ""
+
     command_list = [
         "/sbin/lspci -D", "grep 'Non-Volatile memory controller:'"]
     if ":" in args.nvme:
-        command_list.append("grep '{}'".format(args.nvme.split(":")[1]))
+        vmd_flag = args.nvme.split(":")
+        include_vmd_flag = vmd_flag[1]
+        if include_vmd_flag == "vmd":
+            search_str = "grep 'Non-Volatile memory controller: \\| Volume'"
+            command_list[1] = search_str
+        else:
+            command_list.append("grep '{}'".format(args.nvme.split(":")[1]))
```

Comment:
Do we need to support a model where some CI clusters use VMD and some do not?  If so we probably want a model where only `--nvme=auto` is requested (specified by pipeline-lib), `launch.py` scans for both non-VMD NVMe devices and VMD NVMe devices, and priority is given to VMD NVMe devices when replacing `bdev_list` keywords.

Reply:
Maybe we should meet with Samir and Sydney and understand the requirements here.

Reply:
Well, we can do that probably later, Right now let's have an option so by default all tests with NVMe only run fine as we do run in Ci currently. 

There could be different HW and different numbers of NVMe drives and they may not be connected similar to raiser card or PCIe for VMD. So the first target is to get VMD testing coverage locally and in CI and fix the issue from the DAOS side. We can use the -n auto:vmd option to run existing tests with VMD only on the specific HW which has the VMD enabled and later we can automate the pipeline lib once we see VMD is stable enough and has enough HW to enabled it.